docs/proposals/webhook-validation.md

```diff
@@ -0,0 +1,130 @@
+
+Examples of invalid resources include:
+
+- a BuildRun referencing a Build that doesn't exist
+- a BuildRun referencing a ServiceAccount that doesn't exist
+- a Build referencing a (Cluster)BuildStrategy that doesn't exist
+- a BuildStrategy containing zero steps
+
+(This is a partial list, and new validation can be assumed to be added over time as the Shipwright API evolves.)
+
+By allowing invalid resources to be accepted and stored, we require clients to watch or poll all resources to ensure their request was valid, and we add to the API surface of resources that otherwise would not need to report a status, like a Build.
+
+Always accepting and storing resources, even invalid ones, can also lead to additional storage of known-invalid objects, which clutter the user experience and can lead to confusion, and in extreme cases, cluster resource exhaustion.
+
+Mixing reconciliation and validation code in the controller also complicates our codebase, leading to PRs like https://github.com/shipwright-io/build/pull/641 which add new validation-centric statuses that the user must watch for.
+As we add new validation scenarios, we currently add new `Reason` strings that clients must be aware of, which is cumbersome and won't scale indefinitely.
+
+### Goals
+
+Enforce synchronous validation via a new Deployment and a [ValidatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook), and remove validation-centric logic from the reconciler.
+
+The reconciler should be able to assume in many cases that the resource it is reconciling has already been validated.
+
+### Non-Goals
+
+- Include a mutating or defaulting or conversion webhook.
+  If we decide these are useful, we can add them in future proposals.
+
+## Proposal
+
+Rely on [kubebuilder's webhook support](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html) to add a new webhook validation controller, and include it in Shipwright's default installation as a K8s Deployment.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Certificates 
+
+The webhook deployment will also need to manage certificates since K8s webhook requests are sent over HTTPS.
+Setting up these certificates securely can be cumbersome.
+
+The [Knative](https://knative.dev) project has [some libraries to make this management easier](https://github.com/knative/pkg/tree/main/webhook) by taking care of cert generation, registration and refreshing, but mixing Knative-style components and existing `controller-gen`/KubeBuilder-style components might make complicate development.
+
+#### Webhook Availability
+
+Once registered, if the K8s API server can't reach the defined webhook service, _all_ requests will fail for registered CRD types.
+The webhook deployment is stateless, so it can be easily replicated and scheduled across multiple nodes to ensure availability in the case of node failure.
+
+### Risks and Mitigations
```

Comment:
Can we include here that this will have an impact to existing workflows. For example today workflow X will rely on the Status fields of a Build, tomorrow it should rely on whatever the webhook sends to stdout? (lacking on insights in here)